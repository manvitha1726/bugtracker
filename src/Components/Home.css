.banner-container {
    background-color: rgba(0, 0, 0, 0.6);
    background-image: url('https:/images.unsplash.com/photo-1484417894907-623942c8ee29?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1032&q=80');
    background-repeat: no-repeat;
    background-position: center;
    background-size: cover;
    min-height: 60vh;
    height: 80vh;
    display: flex;
    justify-content: center;
    background-blend-mode: overlay;
    align-items: center;
    color:rgb(247, 238, 238);
}
.row-container {
  display: flex;
  align-items: center;
  border: solid black 1px;
  background-color: rgb(199, 231, 231);
}


.pointer-icon{
    position: absolute;
    right: 10%;

  }

  .pointer-icon:hover{
    cursor: pointer;
  }

  .FaPlusIcon {
    cursor: pointer;
  }

.icon-container {
  display: flex;
  flex-direction: column;
  align-items: center;
  padding-left: 20px;
  padding-top:20px;
   
}

.icon-container p {
  margin-top: 0;
}

.heading-container {
  margin-right: 15%;
  flex-grow: 1;
}

button
{
    border-radius: 10px;
    cursor: pointer;
    background-color:black;
    color:white;
    height: 40px;
    padding:5px;
}
.heading{

    font-family:'Trebuchet MS', 'Lucida Sans Unicode', 'Lucida Grande', 'Lucida Sans', Arial, sans-serif;
    text-shadow: 2px 2px 5px rgb(75, 70, 70);
    color:black;
   
    /* font-size:20px; */
}
.alignright{
    top:5%;
    left:5%;
    position:absolute;
    
}
.align{
    top:5%;
    left:79%;
    position:absolute;
}
.cardu{
   /* height:200px; */
   width:200px;
   background:rgb(139, 200, 209);
   box-shadow: 3px 3px 2px 3px rgb(240, 235, 235);
   font-weight:600;
   padding:20px;
   
}
.addbox{
    width:150px;
}
.space{
    
    width:60px;
    border-width:0px;
    height:30px;
}
.footerposition{
    top:95%;
    position:fixed;
    left:40%;
    
}
.projectmain{
    background-image:url('https://th.bing.com/th/id/OIP.54aDRJHlX7jC15ANjLWg-QHaEo?pid=ImgDet&rs=1') ;
}
.addprojectbt{
    border-width:0px;
    padding:10px;
    background: black;
    color: #eee;
    border-color:white;
    transition: all 0.3s ease;
}
.copyrights{
    color:black;
}