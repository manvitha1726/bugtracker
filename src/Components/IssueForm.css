.fixedwidth {
  width: 250px;
  margin-right: 10px;
}

.fixedwidthtext {
  width: 250px;
  height: 50px;
}

.container {
  width: 60%;
  border-radius: 5px;
  background-color: white;
  padding: 20px;
  align-items: center;
  margin: 20px auto; 
  box-shadow: 2px 2px 5px 0px rgba(0,0,0,0.75);
}

.col-25 {
  float: left;
  width: 35%;
  margin-top: 6px;
  text-align: left;
  padding-left: 8%; 
}

.col-75 {
  float: left;
  width: 50%;
  margin-top: 6px;
}

.row::after {
  content: "";
  display: table;
  clear: both;
  padding: 8px;
}
input:focus,textarea:focus{
  margin:8px 0;
  padding:10px;
  border:1.5px solid #aaa;
  border-radius:4px;
  outline:none;
  transition:.2s;
  box-sizing:border-box;
  border-color: rgb(139, 200, 209);
  box-shadow: 0 0 8px 0 rgb(139, 200, 209)
}
label {
  font-size: 18px;
  cursor: pointer;
}
input,textarea{
  margin:8px 0;
  padding:10px;
}
.options {
  margin-top: 1em;
  display: flex;
  justify-content: space-between;
}
select {
  appearance: none;
  width: 9em;
  padding: 0.30em 0.5em;
  font-size: 1.00em;
  letter-spacing: 0.5px;
  cursor: pointer;
  background-color:rgb(139, 200, 209);
  border: none;
  color: #ffffff;
  background-repeat: no-repeat;
  background-size: 1.0em;
  background-image: url("arrow.png");
  background-position: 7.5em;
  outline: none;
  border-radius: 0.3em;
}
select::-ms-expand {
  display: none;
}
select option {
  background-color: #ffffff;
  color:rgb(139, 200, 209);
  letter-spacing: 0.06em;
  font-weight: 400;
  font-size: 1.12em;
}

.required:before {
  content: "*";
  font-weight: bold;
  color: red;
}