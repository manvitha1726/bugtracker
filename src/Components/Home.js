// import React from 'react';
// import { Link } from 'react-router-dom';
import './Home.css';
import React from 'react';
import Footer from './Footer';


const Home = () => {

    return (
        <div>
            <div className="banner-container">
                <div className="text-center">
                    <h1 className="h1">Welcome to IssueTrackingTool</h1>
                    <br/><br/><br/>
                    <h4 className="h4">Go Ahead and Analyse Issues!</h4>
                    <br/><br/><br/>
                    {/* <center>
                        <a href="/projects">
                    <button className='button-background-color'>
                        Explore Projects
                    </button></a>
                    </center> */}
                </div>
            </div>
            <Footer/>
        </div>
    );
};

export default Home;