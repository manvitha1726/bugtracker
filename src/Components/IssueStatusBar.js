import React, { useEffect, useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import {getProjectById, getProjectNameProjectId} from '../Features/ProjectsSlice';
import { setSelectedFilters, setSelectedIssueId } from '../Features/SelectedFieldsSlice';
import { FaPlus ,FaEye,FaPencilAlt,FaSort, FaImage} from 'react-icons/fa';
import {GetPagesCount,GetIssuesByPagination} from "../Features/IssueSlice";
import {useNavigate} from 'react-router-dom'; 
import Pagination from './Pagination/Pagination';
import './Home.css';
import EmployeeDropdown from './EmployeeDropdown';
import ImageCarouselModal from './ImageCarouselModal.js';

function IssueStatusBar() {     
    const dispatch = useDispatch();  
    const navigate = useNavigate();
    const [searchTerm, setSearchTerm] = useState('');
    const [currentPage, setCurrentPage] = useState(1);
    const [postsPerPage, setPostsPerPage] = useState(4);
    const [sortOrder, setSortOrder] = useState('');
    const [filteredData,setFilteredData]=useState();
    const [dataLoaded, setDataLoaded] = useState(false);
    const [currentPosts, setCurrentPosts] = useState([]);
    const [dataSorted, setDataSorted] = useState(false);
    const [isDataFiltered, setIsDataFiltered] = useState(false);
    const [selectedAssignedEmployee, setSelectedAssignedEmployee] = useState(-1);
    const [IdentifiedEmployee, setIdentifiedEmployee] = useState(-1);
    const [isFromLandingPage, setIsFromLandingPage] = useState(true);
    const projectname = useSelector((state) => state.projects.ProjectName.projectname);
    const {paginationdata, loading, error } = useSelector((state) => state.issues);
    const noOfpages=useSelector((state)=>state.issues.pagesCount)
    

    const [issueFilterVal, setIssueFilterVal] = useState({
      status: 'Any',
      identfiedemp:-1,
      assignTo: -1,
      priority: 'Any',
      seviority: 'Any'
    });

    const ProjectId = useSelector((state) => state.selectedFields.selectedProjectId);
    const selectedFilters = useSelector((state) => state.selectedFields.selectedFilters);
    // console.log("projectidd",ProjectId);

    useEffect(() => { 
      const { status, identfiedemp, assignTo, priority, seviority } = issueFilterVal;
      dispatch(GetPagesCount({ProjectId, status, identfiedemp, assignTo, priority, seviority, postsPerPage}));
      dispatch(getProjectNameProjectId(ProjectId))
    },[])

    useEffect(() => { 
      const { status, identfiedemp, assignTo, priority, seviority } = issueFilterVal;
      // console.log(issueFilterVal, selectedAssignedEmployee, IdentifiedEmployee)
      dispatch(GetIssuesByPagination({ProjectId, status, identfiedemp, assignTo, priority, seviority, postsPerPage,currentPage}));
    },[currentPage])

    // console.log("jiohhvu",paginationdata,noOfpages,currentPage);

    useEffect(() => {
      console.log("selected filters in ISB--", selectedFilters);
      handleFiltersFromLandingPage();
      handleFilterApply();
    }, []);

    
      
    useEffect(() => {
      // console.log("asd", isFromLandingPage);
      if(!isFromLandingPage && paginationdata.length>0){ 
        const filteredData1 = paginationdata.filter(issue => {
              const lowerCaseissueId = issue.issueId.toLowerCase();
              const lowerCaseShortDescription = issue.shortDescription.toLowerCase();
              const lowerCaseStatus = issue.status.toLowerCase();
              const lowerCasePriority = issue.priority.toLowerCase();
              const lowerCaseSearchTerm=searchTerm.toLowerCase();
            
              return (
                lowerCaseShortDescription.includes(lowerCaseSearchTerm) ||
                lowerCaseStatus.includes(lowerCaseSearchTerm) ||
                lowerCasePriority.includes(lowerCaseSearchTerm) ||
                lowerCaseissueId.includes(lowerCaseSearchTerm)
              );
          
        }); 
      setFilteredData(filteredData1)
      setDataLoaded(true);
      setCurrentPosts(filteredData1);
    }
    else{
      setCurrentPage(1);
    }
    // handleFilterApply();
    }, [searchTerm])  

    useEffect(() => {
      if(paginationdata.length > 0){
        setFilteredData(paginationdata)
        setDataLoaded(true);
        setCurrentPosts(paginationdata);
      }
      else if(noOfpages == 0){
        setFilteredData([]);
        setDataLoaded(true);
        setCurrentPosts([]);
      }
      else{
        setCurrentPage(1);
        setFilteredData(paginationdata);
        setDataLoaded(true);
        setCurrentPosts(paginationdata);
      }
    }, [paginationdata])

    // console.log("filteredData",filteredData);
    useEffect(() =>{
      if(dataLoaded){
      // const lastPostIndex = currentPage * postsPerPage;
      // const firstPostIndex = lastPostIndex - postsPerPage;
      setCurrentPosts(filteredData);
      setDataSorted(false)
      }
    }, [dataSorted, currentPage])

    useEffect(() =>{
      if(dataLoaded){
      // const lastPostIndex = currentPage * postsPerPage;
      // const firstPostIndex = lastPostIndex - postsPerPage;
      setCurrentPosts(filteredData);
      setIsDataFiltered(false);
      }
    }, [isDataFiltered, currentPage])


    useEffect(() => {
      setIssueFilterVal((prevFilters) => ({ ...prevFilters, assignTo: `${selectedAssignedEmployee}` }));
    }, [selectedAssignedEmployee]);
  
    useEffect(() => {
      setIssueFilterVal((prevFilters) => ({ ...prevFilters, identfiedemp: `${IdentifiedEmployee}` }));
    }, [IdentifiedEmployee]);


    const handleFiltersFromLandingPage = () => {
      if(selectedFilters!==null){
        setIssueFilterVal(selectedFilters)
        // setFiltersFromLandingPage(true);
      }
    }
    // const disPatchIssues=(page)=>{
    //   // const { status, identfiedemp, assignTo, priority, seviority } = issueFilterVal;
    //   setCurrentPage(page);
    //   // console.log("vcjsja",ProjectId, status, identfiedemp, assignTo, priority, seviority, postsPerPage,currentPage)
    //   // dispatch(GetIssuesByPagination(ProjectId, status, identfiedemp, assignTo, priority, seviority, postsPerPage,currentPage));
    // }
    const handleSort = () => {
        const sortedData = [...filteredData].sort((a, b) => {
        const priorityOrder = { P1: 1, P2: 2, P3: 3 };
        return priorityOrder[a.priority] - priorityOrder[b.priority];
      });
      if(sortOrder == "asc"){
        sortedData.reverse();
        setSortOrder("dsc");
      }
      else{
        setSortOrder("asc");
      }
      setFilteredData(sortedData);
      setDataSorted(true)
    };
     
    const handleStatusSort = () => {
      const sortedData = [...filteredData].sort((a, b) => {
      const priorityOrder = { Open: 1, Hold: 2, "In Progress": 3 , Close: 4};
        return priorityOrder[a.status] - priorityOrder[b.status];
      });
      if(sortOrder == "asc"){
        sortedData.reverse();
        setSortOrder("dsc");
      }

      else{
        setSortOrder("asc");
      }
      setFilteredData(sortedData);
      setDataSorted(true)
    };

    const handleSevioritySort = () => {
      const sortedData = [...filteredData].sort((a, b) => {
      const seviorityOrder = { S1: 1, S2: 2, S3: 3 , S4: 4};
        return seviorityOrder[a.seviority] - seviorityOrder[b.seviority];
      });
      if(sortOrder == "asc"){
        sortedData.reverse();
        setSortOrder("dsc");
      }

      else{
        setSortOrder("asc");
      }
      setFilteredData(sortedData);
      setDataSorted(true)
    };
    const handlePlusIconClick = () => {
      navigate(`/projects/${ProjectId}/AddIssue`);
    };
    const handleEditIcon = (issueId) => {
      dispatch(setSelectedIssueId(issueId));
      navigate(`/projects/${ProjectId}/EditIssue${issueId}`);
    };
  
  
     const handleFilterChange = (event) => {
      const { name, value } = event.target;
      setIssueFilterVal((prevFilters) => ({ ...prevFilters, [name]: value }));
     }
     const handleFilterApply = () => {
      // console.log("selected filters --:- ", selectedFilters);
      if(selectedFilters!==null){
        dispatch(GetPagesCount({ProjectId, ...selectedFilters, postsPerPage}));
        dispatch(GetIssuesByPagination({ProjectId, ...selectedFilters, postsPerPage, currentPage}));
        dispatch(setSelectedFilters(null));
      }
      else{
        dispatch(GetPagesCount({ProjectId, ...issueFilterVal, postsPerPage}));
        if(currentPage > 1){
          dispatch(GetIssuesByPagination({ProjectId, ...issueFilterVal, postsPerPage,currentPage:1}));  
        }
        else{
          dispatch(GetIssuesByPagination({ProjectId, ...issueFilterVal, postsPerPage,currentPage}));  
        }
      }
     }

     const NavigateToSelectedIssue = (issueId) => {
      // console.log(issueId);
      dispatch(setSelectedIssueId(issueId));
      navigate(`/projects/${ProjectId}/display-issue${issueId}`);
     }

     const handleSearch = (event) => {
      setSearchTerm(event.target.value)
      setIsFromLandingPage(false);
     }

     const handleFilterReset = () => {
        // console.log("abxha",paginationdata);
        dispatch(GetPagesCount({ProjectId, status:"Any",identfiedemp: -1, assignTo:-1,priority: "Any",seviority: "Any", postsPerPage}));
        dispatch(GetIssuesByPagination({ProjectId, status:"Any",identfiedemp: -1, assignTo:-1,priority: "Any",seviority: "Any", postsPerPage,currentPage}));
        setFilteredData(paginationdata);
        setIsDataFiltered(true);
        setCurrentPage(1);
        setIssueFilterVal({status:"Any",identfiedemp: -1, assignTo:-1,priority: "Any",seviority: "Any"});
     }

  if(loading){
    return (
      <div class="text-center my-auto">
        <br/><br/><br/><br/><br/><br/><br/><br/>
        <div class="spinner-border" role="status">
        </div>
        <br/>
        <span>Loading....</span>
      </div>
    )
   }

  if(error){
    return <h2>Oops Something wrong..</h2>
   }
   
  if(dataLoaded){
      return (
        <div className='Mains-Container'>
         <div className="row-container">
              
                <div className="icon-container">
                  <FaPlus className="icon rounded p-1" style={{ backgroundColor: "black",height:'25px',width:'25px',color:'white',marginLeft:'80px',}} onClick={handlePlusIconClick} />
                  <p style={{color:'black ',marginLeft:'60px'}}>Add Issue</p>
                
              </div>
              <div className='heading-container'>
                <h3> {projectname} Issues</h3>
              </div>
              <div className='align'>
                <input className="pa2 bb br3 ma2 shadow" type="text" placeholder="Search Issue" value={searchTerm} onChange={handleSearch} />
              </div>
          </div>



          <div className='filters' >
            <h5  style={{"font-family":"'Trebuchet MS'"}}>Filter</h5>
              
              <div className='filter-row' style={{display:'flex', flexDirection:'row'}}>
                <div className='each-filter' style={{display:'flex', flexDirection:'column',marginRight:"20px"}}>
                    <label>Status</label>
                    <select 
                    className='IssueStatusBar-background-color'
                      name='status'
                      value={issueFilterVal.status}
                      onChange={handleFilterChange}
                    >
                      <option value="Any">Any</option>
                      <option value="Open">Open</option>
                      <option value="Close">Closed</option>
                      <option value="In Progress">In Progress</option>
                      <option value="Hold">Hold</option>
                    </select>
                </div>
                &nbsp;&nbsp;&nbsp;
                <div className='each-filter' style={{display:'flex', flexDirection:'column',marginRight:"20px"}}>
                    <label>Priority</label>
                    <select 
                    className='IssueStatusBar-background-color'
                      name='priority'
                      value={issueFilterVal.priority|| 'Any'}
                      onChange={handleFilterChange}
                    >
                      <option value="Any">Any</option>
                      <option value="P1">P1</option>
                      <option value="P2">P2</option>
                      <option value="P3">P3</option>
                    </select>
                </div>
                &nbsp;&nbsp;&nbsp;
                <div className='each-filter' style={{display:'flex', flexDirection:'column',marginRight:"20px"}}>
                    <label>Severity</label>
                    <select 
                      className='IssueStatusBar-background-color'
                      name='seviority'
                      value={issueFilterVal.seviority|| 'Any'}
                      onChange={handleFilterChange}
                    >
                      <option value="Any">Any</option>
                      <option value="S1">S1</option>
                      <option value="S2">S2</option>
                      <option value="S3">S3</option>
                      <option value="S4">S4</option>
                    </select>
                </div>
                &nbsp;&nbsp;&nbsp;
                <div className='each-filter' style={{display:'flex', flexDirection:'column',marginRight:"20px"}}>
                    <label>Identfied by</label>
                    <EmployeeDropdown isFromFilters={true} callBackFunc={setIdentifiedEmployee} />
                </div>
                &nbsp;&nbsp;&nbsp;
                <div className='each-filter' style={{display:'flex', flexDirection:'column',marginRight:"20px"}}>
                    <label>Assigned Employee</label>
                    <EmployeeDropdown isFromFilters={true} callBackFunc={setSelectedAssignedEmployee} employeeFromFiltersLandingPage={issueFilterVal.assignTo} />
                </div>                
              </div>
              <br />
              <div>
              <button className='button-background-color' onClick={handleFilterApply}>Apply Filters</button>
              &nbsp;&nbsp;&nbsp;
              <button className='button-background-color' onClick={handleFilterReset}>Reset</button>
              </div>
          
          </div>

          <div className='Issue-table'>
          <table className="table table-bordered rounded-lg">
            <thead>
              <tr>
                <th className='p-3 text-center' style={{backgroundColor:"rgb(199, 206, 207)", width:'20%'}}>Issue Id</th>
                <th className='p-3 text-center'  style={{backgroundColor:"rgb(199, 206, 207)"}}>Status &nbsp; <FaSort onClick={handleStatusSort}/></th>
                <th className='p-3 text-center' style={{backgroundColor:"rgb(199, 206, 207)"}}>Priority &nbsp;<FaSort onClick={handleSort}/></th>
                <th className='p-3 text-center'  style={{backgroundColor:"rgb(199, 206, 207)"}}>Severity &nbsp;<FaSort onClick={handleSevioritySort}/></th>
                <th className='p-3 text-center'  style={{backgroundColor:"rgb(199, 206, 207)"}}>Category</th>
                <th className='p-3 text-center'  style={{backgroundColor:"rgb(199, 206, 207)", width:'25%'}}>Summary</th>
              </tr>
            </thead>
            <tbody>
              {currentPosts.map(issue => ( 
                <tr key={issue.issueId}>
                  <td className='p-3 table-1stcol' style={{position:"relative"}}>
                    
                  <div className='row-'>
                        <a onClick={() => NavigateToSelectedIssue(issue.issueId)} className='clickable-'>
                            {issue.issueId}
                        </a> &nbsp;&nbsp;&nbsp;
                        <div className='pointer-icon1'>
                            <ImageCarouselModal images={issue.images} />
                        </div>
                        <FaPencilAlt className='pointer-icon2' onClick={() => handleEditIcon(issue.issueId)}/>
                    </div>
                  </td>
                  <td className='p-3'> 
                    {issue.status}
                    {/* <center>
                    <select 
                      value={issue.status|| 'Open'}
                      onChange={e => handleStatusChange(issue.issueId, e.target.value)}
                    >
                      <option value="Open">Open</option>
                      <option value="Close">Closed</option>
                      <option value="In Progress">In Progress</option>
                      <option value="Hold">Hold</option>
                    </select>
                    </center>  */}
                  </td>
                  <td className='p-3'>
                    {issue.priority}
                  </td>
                  
                  <td >
                    {issue.seviority}
                  {/* <center>
                  <FaEye className='pointer-icon' onClick={() => handleViewIcon(issue.issueId)} /></center> */}
                  </td>
                  <td >
                    {issue.category}
                  </td>
                  <td>
                    {issue.shortDescription}
                  </td>
                  
                </tr>
              ))}
              
              </tbody>
            </table>
            <Pagination
                  noOfpages={noOfpages}
                  setCurrentPage={setCurrentPage} 
                  currentPage={currentPage}
                />
          </div> 
        </div>
      );
  }
}

export default  IssueStatusBar; 