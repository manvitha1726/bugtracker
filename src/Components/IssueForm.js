import React, { useEffect, useState } from "react";
import { AddNewIssue } from "../Features/IssueSlice";
import { useDispatch ,useSelector} from "react-redux";
import { useNavigate } from 'react-router-dom';
import EmployeeDropdown from "./EmployeeDropdown";
import AddEmployee from "./AddEmployee";
import ImageUpload from "./ImageUpload/ImageUpload";
import validateForm from './formValidation';
import { getAllProjects } from "../Features/ProjectsSlice";
import './IssueForm.css';

function IssueForm() {
  const navigate = useNavigate();
  const dispatch = useDispatch();
  const projectId = useSelector((state) => state.selectedFields.selectedProjectId);
  const projObj= useSelector((state) => state.projects);
  const initialFormData = {
    projectId: `${projectId}`,
    issueName: "",
    issueType: "",
    moduleName: "",
    description: "",
    summary: "",
    identfiedemp: "",
    priority: "Low",
    targetdate: "",
    assignTo: "",
    progressreport: "",
    stepsToReproduce: "",
    testingType: "Smoke Testing",
    iterationNumber: "",
    seviority: 'S1',
    linkToPast: null,
    images: ""
  
  };

  const [val1, setVal1] = useState(1);
  const [formData, setFormData] = useState(initialFormData);
  const [selectedIssue, setSelectedIssue] = useState("Bug");
  const [selectedSeviority, setSelectedSeviority] = useState("S1");
  const [selectedTesting, setSelectedTesting] = useState("Smoke Testing");
  const [selectedPriority, setSelectedPriority] = useState('Low');
  const [selectedAssignedEmployee, setSelectedAssignedEmployee] = useState();
  const [IdentifiedEmployee, setIdentifiedEmployee] = useState();
  const [attachedFiles, setAttachedFiles] = useState('');
  const [errors,setErrors]= useState({});
  const [issues,setIssues]= useState([]);

  useEffect(() => {
    setFormData((prevFormData) => ({ ...prevFormData, assignTo: `${selectedAssignedEmployee}` }));
    // setFormData((prevFormData) => ({ ...prevFormData, assignTo: `${selectedAssignedEmployee}` }));
  }, [selectedAssignedEmployee]);

  useEffect(() => {
    setFormData((prevFormData) => ({ ...prevFormData, identfiedemp: `${IdentifiedEmployee}` }));
  }, [IdentifiedEmployee]);
  useEffect(() => {
    console.log("image from form: ", attachedFiles);
    setFormData((prevFormData) => ({ ...prevFormData, images: attachedFiles }));
  }, [attachedFiles]);

  useEffect(() => {
    setFormData((prevFormData) => ({ ...prevFormData,  issueType: selectedIssue}));
    }, [selectedIssue])

  useEffect(() => {
    dispatch(getAllProjects());
    console.log("projects data",projObj.data);
  }, []);

  const NavigateBackClick1 = () => {
    navigate(`/projects/${projectId}`);
  };

  const NavigateToIssues = () => {
    navigate(`/projects/${projectId}`);
  }
  const handleSelectedSeviority = (event) => {
    setSelectedSeviority(event.target.value);
    setFormData((prevFormData) => ({ ...prevFormData, seviority: event.target.value }));
  }

  const handleIssueSelection = (event) => {
    setSelectedIssue(event.target.value);
    setFormData((prevFormData) => ({ ...prevFormData, issueType: event.target.value }));
  };

  const handleTestingSelection = (event) => {
    setSelectedTesting(event.target.value);
    setFormData((prevFormData) => ({ ...prevFormData, testingType: event.target.value }));
  };

  const handleSelectedPriority = (event) => {
    setSelectedPriority(event.target.value);
    setFormData((prevFormData) => ({ ...prevFormData, priority: event.target.value }));
  };

  const handleChange = (event) => {
    const { name, value } = event.target;
    setFormData((prevFormData) => ({ ...prevFormData, [name]: value }));
  };


  const handleSubmit = (event) => {
    const validationData={
            issueName:formData.issueName,
            moduleName:formData.moduleName,
            summary:formData.summary,
            identfiedemp:formData.identfiedemp,
            targetdate:formData.targetdate,
            progressreport:formData.progressreport,
            stepsToReproduce:formData.stepsToReproduce,
            description:formData.description,
            iterationNumber:formData.iterationNumber
    }
    event.preventDefault();
    console.log("Before submit validation data", validationData);
    const formErrors = validateForm(
      validationData
    );
    console.log("form errors",formErrors)
    if (formErrors){
      setErrors(formErrors);
      return;
    }
    setIssues([...issues, formData]);
    dispatch(AddNewIssue(formData))
    .then((response) => {
      console.log("After Issue form submitted response is : ",response);
      NavigateToIssues();
    })
    .catch(error => {
        console.error('Error updating bug status:', error);
      });
  };
 return (
  <>
        <form className="container">
        <h3 className="text-center">Enter Issue Details: {projObj.data[projectId-1].projectname}</h3><br/>
        <div class="row">
    <div class="col-25">
      <label  className="form-label" for="inputEmail4">Issue Name</label> 
      <input type="text" className="form-control" id="inputEmail4" name="issueName" onChange={handleChange} placeholder="Issue Name"/>
    </div>
    <div class="col-75">
    <label className="form-label" for="inputEmail4">Module Name</label> 
      <input type="text" className="form-control" id="inputEmail4" name="moduleName" onChange={handleChange} placeholder="Module Name"/>
    </div>
    </div>
      <div class="row">
        <div class="col-3">
            <label className="form-label">Issue Type</label> 
              <select className="drop" id="IssueType" value={selectedIssue} onChange={handleIssueSelection} required>
                      <option value="Bug">Bug</option>
                      <option value="Defect">Defect</option>
                    </select>
        </div>
        <div class="col-3">
          <label className="form-label">Identified Employee</label>
          <EmployeeDropdown val={val1} callBackFunc={setIdentifiedEmployee} prjID={projectId} />
                <div className="validations">
                  {errors.identfiedemp && <span>{errors.identfiedemp}</span>}
                </div>
        </div>
        <div class="col-3">
        <label className="form-label" htmlFor="priority">Priority</label>
                <select id="IssueType" value={selectedPriority} onChange={handleSelectedPriority} required>
                  <option value="Low">Low</option>
                  <option value="Medium">Medium</option>
                  <option value="High">High</option>
                </select>
        </div>
        <div className="col-3">
            <label className="form-label" htmlFor="priority">Seviority</label>
            <select id="seviority" value={selectedSeviority} onChange={handleSelectedSeviority} required>
              <option value="S1">S1</option>
              <option value="S2">S2</option>
              <option value="S3">S3</option>
              <option value="S4">S4</option>
            </select>
          </div>
      </div>
        <div class="row">
    <div className="col-25">
           <label className="form-label" htmlFor="targetdate">Target Date</label> 
          <input type="date" className="form-control" id="targetdate" name="targetdate" value={formData.targetdate} onChange={handleChange}/>
          <div className="validations">
            {errors.targetdate && <span>{errors.targetdate}</span>}
          </div>
          </div>
  </div>

        <div className="row">
          <div className="col-25">
            <label className="form-label" htmlFor="progressreport">Progress Report</label> 
            <input type="text" className="form-control" id="progressreport" placeholder="Progress Report" name="progressreport" value={formData.progressreport} onChange={handleChange}/>
            <div className="validations">
              {errors.progressreport && <span>{errors.progressreport}</span>}
            </div>
          </div>
          <div className="col-75">
            <label className="form-label" htmlFor="stepsToReproduce">Steps To Reproduce</label>
            <input type="text" className="form-control" id="stepsToReproduce" placeholder="StepsTo Reproduce" name="stepsToReproduce" value={formData.stepsToReproduce} onChange={handleChange}/>
            <div className="validations">
              {errors.stepsToReproduce && <span>{errors.stepsToReproduce}</span>}
            </div>
            </div>
        </div>

        <div className="row">
            <div className="col-25">
                <label className="form-label" htmlFor="iterationNumber">Iteration Number</label> 
              <input type="number" className="form-control" id="iterationNumber"  name="iterationNumber" placeholder="Iteration Number" value={formData.iterationNumber} onChange={handleChange} />
            
              <div className="validations">
                  {errors.iterationNumber && <span>{errors.iterationNumber}</span>}
              </div>
            </div>
            
            &nbsp;&nbsp;&nbsp;&nbsp;
              &nbsp;&nbsp;&nbsp;&nbsp;
            <div className="col-50">
                <label className="form-label" htmlFor="testingType">Testing Type</label>
                <select id="testingType" style={{ height:" 30%"}} value={selectedTesting} onChange={handleTestingSelection}>
                  <option value="Smoke Testing">Smoke Testing</option>
                  <option value="Regression Testing">Regression Testing</option>
                </select>
            </div>
            <div className="col-50">
                <label className="form-label" htmlFor="assignTo">Assigned To</label>
                <br/>
                <EmployeeDropdown val={val1} callBackFunc={setSelectedAssignedEmployee} />
              <AddEmployee func={setVal1} projectId={projectId} />
                
            </div>
        </div>
        <br/><br/>

        <div className="row">
          {/* <div className="col-25">
            <label className="form-label" htmlFor="summary">Summary</label>
          </div> */}
          <div className="col">
            <textarea cols={90} id="summary" placeholder="Summary" name="summary" value={formData.summary} onChange={handleChange} required />
            <div className="validations">
              {errors.summary && <span>{errors.summary}</span>}
            </div>
            </div>
        </div>
      
        <div className="row">
          {/* <div className="col-25">
            <label className="form-label" htmlFor="summary">Summary</label>
          </div> */}
          <div className="col">
            <textarea cols={90} id="description" placeholder="Description" name="description" value={formData.Description} onChange={handleChange}/>
            <div className="validations">
              {errors.description && <span>{errors.description}</span>}
            </div>
          </div>
        </div>
        
        <div className="row" style={{display:'flex', flexDirection:'row'}}>
            <label className="form-label" htmlFor="images">Upload Image</label>
            <ImageUpload callBackFunc={setAttachedFiles} />
        </div>

        <center>
            <button type="submit" onClick={handleSubmit}>Add Issue</button> 
            &nbsp;&nbsp;
            <button onClick={NavigateBackClick1}>Cancel</button>
        </center>
        </form>
        </>
        );

}
      

export default IssueForm;