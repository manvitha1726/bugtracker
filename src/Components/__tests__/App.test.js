import { render, screen,fireEvent } from '@testing-library/react';
import App from '../../App';
import Home from '../Home';
import IssueForm from '../IssueForm';
import IssueStatusBar from '../IssueStatusBar';


describe('Rendering of components ', () => {

  test('renders learn react link', () => {
    render(<Home />);
    const linkElement = screen.getByText(/Explore Projects/i);
    expect(linkElement).toBeInTheDocument();
    });

  // test('should handle button click event', () => {
  //   render(<IssueStatusBar />);
    
  //   const button = screen.getBytext('Add Issue');
  //   fireEvent.click(button);
      
  //   expect(/*Issue Name*/).toBeInTheDocument();
  //   });
  
});

describe('creating snapshorts of components ', () => {

  test('Home Component matches snapshot', () => {
      const { asFragment } = render(<Home />);
      expect(asFragment()).toMatchSnapshot();
    });

});
