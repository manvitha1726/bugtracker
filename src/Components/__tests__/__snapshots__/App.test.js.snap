// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`creating snapshorts of components  Home Component matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      data-testid="header-element"
    >
      <h1
        class="heading text-center"
      >
        Issue Tracking Tool
      </h1>
    </div>
    <div
      class="banner-container"
    >
      <div
        class="text-center"
      >
        <h1
          class="h1"
        >
          Welcome to IssueTrackingTool
        </h1>
        <br />
        <br />
        <br />
        <h4
          class="h4"
        >
          Go Ahead and Analyse Issues!
        </h4>
        <br />
        <br />
        <br />
        <center>
          <a
            href="/projects"
          >
            <button>
              Explore Projects
            </button>
          </a>
        </center>
      </div>
    </div>
    <div
      class="bg-[#FA7D19] text-white"
    >
      <footer
        class="footer p-10 justify-items-center"
        data-testid="footer"
      />
      <div
        class="divv"
      >
        <center>
          <p
            class="copyrights"
          >
            Copyright Â© 2023 - All rights reserved by Syren
          </p>
        </center>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
